pkg/rpcclient/rolemgmt/roles.go

```diff
@@ -98,3 +103,47 @@ func (c *Contract) DesignateAsRoleTransaction(role noderoles.Role, pubs keys.Pub
 func (c *Contract) DesignateAsRoleUnsigned(role noderoles.Role, pubs keys.PublicKeys) (*transaction.Transaction, error) {
 	return c.actor.MakeUnsignedCall(Hash, designateMethod, nil, int(role), pubs)
 }
+
+// FromStackItem converts provided [stackitem.Array] to DesignationEvent or returns an
+// error if it's not possible to do to so.
+func (e *DesignationEvent) FromStackItem(item *stackitem.Array) error {
+	if item == nil {
+		return errors.New("nil item")
+	}
+
+	arr, ok := item.Value().([]stackitem.Item)
+	if !ok {
+		return errors.New("not an array")
+	}
+	if len(arr) != 2 {
+		return errors.New("wrong number of event parameters")
+	}
+
+	roleBigInt, err := arr[0].TryInteger()
+	if err != nil {
+		return fmt.Errorf("invalid role: %w", err)
+	}
+	if !roleBigInt.IsUint64() {
+		return fmt.Errorf("role overflow: %s", roleBigInt.String())
+	}
+	role := roleBigInt.Uint64()
+	if role > math.MaxUint8 {
+		return fmt.Errorf("role overflow: %d > %d", role, math.MaxUint8)
+	}
+	e.Role = noderoles.Role(byte(role))
```

Comment:
What I'm thinking about is: let's add `noderoles.IsValid` definition. Use `noderoles.StateValidator` and `noderoles.last` as lower/upper bound of a valid role. Reuse this helper from native Designation instead of the one presented there:
https://github.com/nspcc-dev/neo-go/blob/fe0a7589520eda61f07f6f5f73e5739125179431/pkg/core/native/designate.go#L98-L101

Use `noderoles.IsValid` from this `FromStackItem` method. Also add a unit-test for `noderoles.IsValid`.